### PR TITLE
memory: Add Zep Cloud integration

### PR DESCRIPTION
**Pull request by paul-paliychuk**
### PR Checklist

- [x] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [x] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [x] Name your Pull Request title clearly, concisely, and prefixed with the name of the primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `Fixes #123`).
- [x] Describes the source of new concepts.
- [x] References existing implementations as appropriate.
- [x] Contains test coverage for new functions.
- [x] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.

Creates a zep package that aims to integrate with zep cloud memory, link to our [docs](https://help.getzep.com/chat-history-memory/overview). I set up zep.ChatMessageHistory and zep.Memory following your BufferMemory implementation. This can be used by users who want access to long term memory and provide good context to their llm prompts. After each message (or memory) is added to a session, we generate summaries and extract facts from the conversation. When we get memory, we inject the most relevant summary and conversation facts into a system prompt.

### CODE REVIEW

**Comment by paul-paliychuk** [comment]
@tmc Hi, Zep Engineer here. we currently support [TS](https://github.com/getzep/zep-js/tree/main/src/langchain) and [Python](https://github.com/getzep/zep-python/tree/main/zep_python/langchain) LangChain integrations, and have recently released our go-sdk. Excited to deliver LangChain integration for our golang users too! Please let me know if you have any questions/suggestions about the implementation. Would love to hear how langchaingo users might prefer to use our memory. 

For now I set up an example using ConversationChain, but we have a bunch more stuff we'd like to port to Go.

**Comment by paul-paliychuk** [comment]
Also I notice that the build examples is failing, this is due to /memory/zep import not being available yet.

**Comment by devalexandre** [comment]
> Also I notice that the build examples is failing, this is due to /memory/zep import not being available yet.

for example don't need use go.mod , can import only package

**Comment by paul-paliychuk** [comment]
@devalexandre Hi, removed go.mod from example project, also upgraded zep sdk version with more straightforward initialization.

**Review by devalexandre — APPROVED** [review]
amazing !!!

**Comment by devalexandre** [comment]
@tmc could review it?

**Comment by devalexandre** [comment]
@paul-paliychuk why update to draft?

**Comment by tmc** [comment]
@paul-paliychuk LMK when this is ready for review (or is itnow?)

**Comment by paul-paliychuk** [comment]
@tmc apologies, we were refactoring our go sdk api recently, will update PR components soon and mark as ready to review. Thanks!

**Comment by paul-paliychuk** [comment]
@tmc @devalexandre Ready for review

File: memory/zep/zep_chat_history.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
+package zep
```
Comment (tmc) [inline memory/zep/zep_chat_history.go:1]: please include a doc.go for new packages.

**Comment by tmc** [comment]
Sorry about the delay here, this looks great! A small comment but I think this can go in as-is and be fixed separately.

**Comment by paul-paliychuk** [comment]
@tmc Appreciate the review, will open a docs PR this week

**Pull request merged**